src/infrasys/time_series_metadata_store.py

```diff
@@ -493,40 +577,27 @@ class TimeSeriesCounts:
     time_series_type_count: dict[tuple[str, str, str, str], int]
 
 
-def _make_user_attribute_filter(user_attributes: dict[str, Any], params: list[str]) -> str:
-    attrs = _make_user_attribute_dict(user_attributes)
+def _make_features_filter(features: dict[str, Any], params: list[str]) -> str:
+    attrs = _make_features_dict(features)
     items = []
     for key, val in attrs.items():
-        items.append(f"metadata->>'$.user_attributes.{key}' = ? ")
+        items.append(f"features ->> '$.{key}' = ? ")
         params.append(val)
     return "AND ".join(items)
 
 
-def _make_user_attribute_hash_filter(attribute_hash: str, params: list[str]) -> str:
-    params.append(attribute_hash)
-    return "user_attributes_hash = ?"
-
-
-def _make_user_attribute_dict(user_attributes: dict[str, Any]) -> dict[str, Any]:
-    return {k: user_attributes[k] for k in sorted(user_attributes)}
-
-
-def _compute_user_attribute_hash(user_attributes: dict[str, Any]) -> str | None:
-    if not user_attributes:
-        return None
-
-    attrs = _make_user_attribute_dict(user_attributes)
-    return _compute_hash(bytes(json.dumps(attrs), encoding="utf-8"))
-
-
-def _compute_hash(text: bytes) -> str:
-    hash_obj = hashlib.sha256()
-    hash_obj.update(text)
-    return hash_obj.hexdigest()
+def _make_features_dict(features: dict[str, Any]) -> dict[str, Any]:
+    return {k: features[k] for k in sorted(features)}
 
 
 def _deserialize_time_series_metadata(text: str) -> TimeSeriesMetadata:
     data = json.loads(text)
     type_metadata = SerializedTypeMetadata(**data.pop(TYPE_METADATA))
     metadata = deserialize_value(data, type_metadata.fields)
     return metadata
+
+
+def make_features_string(features: dict[str, Any]) -> str:
+    """Serializes a dictionary of features into a sorted string."""
+    data = dict(sorted(features.items()))
+    return json.dumps(data)
```

Comment:
While Python has a guarantee on preserving dictionary insertion order, I don't think JSON does. That's why I wrote the Julia version like this:
```julia
function make_features_string(features::Dict{String, Union{Bool, Int, String}})
    key_names = sort!(collect(keys(features)))
    data = [Dict(k => features[k]) for k in key_names]
    return JSON3.write(data)
end
```
You'll need to match it.

Reply:
From that code I think you are writing an array and not a dictionary, is that right?

Reply:
why do we need the array?  I do not think it is necessary for indexing

Reply:
It allows us to do perform SQL queries with string comparisons. If we use a JSON object instead of array, field order is non-deterministic.